src/averageAges.js

```diff
@@ -55,7 +65,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const parents = people.filter(person => (onlyWithSon)
+    ? person.sex === 'm' && people.find(
+      mother => person.mother === mother.name)
+    : people.find(mother => person.mother === mother.name)
+  );
+
+  const diffAge = parents.map(person => person.born
+    - people.find(mother => person.mother === mother.name).born);
```

Comment:
`find(mother => person.mother === mother.name).born` you may pass this logic into separate function `getMotherBornYear` or smth like this

Reply:
Not fixed.
You can pass person.mother to your function as an argument and use it inside your function as before.